WordPress/src/main/java/org/wordpress/android/util/SnackbarSequencer.kt

```diff
@@ -0,0 +1,114 @@
+package org.wordpress.android.util
+
+import android.app.Activity
+import android.content.Context
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.wordpress.android.modules.BG_THREAD
+import org.wordpress.android.modules.UI_THREAD
+import org.wordpress.android.ui.utils.UiHelpers
+import org.wordpress.android.util.AppLog.T
+import org.wordpress.android.widgets.WPSnackbar
+import org.wordpress.android.widgets.WPSnackbarWrapper
+import java.util.LinkedList
+import java.util.Queue
+import javax.inject.Inject
+import javax.inject.Named
+import javax.inject.Singleton
+import kotlin.coroutines.CoroutineContext
+
+const val DURATION_EXTRA_MARGIN = 500L
+
+private const val QUEUE_SIZE_LIMIT: Int = 10
+
+@Singleton
+class SnackbarSequencer @Inject constructor(
+    private val uiHelper: UiHelpers,
+    private val wpSnackbarWrapper: WPSnackbarWrapper,
+    @Named(BG_THREAD) private val bgDispatcher: CoroutineDispatcher,
+    @Named(UI_THREAD) private val mainDispatcher: CoroutineDispatcher
+) : CoroutineScope {
+    private var job: Job = Job()
+
+    private val snackBarQueue: Queue<SnackbarSequencerInfo> = LinkedList()
+
+    override val coroutineContext: CoroutineContext
+        get() = bgDispatcher + job
+
+    fun enqueue(item: SnackbarSequencerInfo) {
+        synchronized(this@SnackbarSequencer) {
```

Comment:
I'd consider using `launch(mainThread)` here instead since using the main thread would ensure synchronous access and I think it makes it easier to understand. There is also no heavy work necessary on the thread. It would mean replacing the next launch with `withContext(bgThread)`

Reply:
Ohh, great idea! I was curious so I tested it and it works as expected.

Reply:
Here is a patch in case you want to use it @develric - https://cloudup.com/cdAXfGQft6N

Reply:
Makes sense, thanks for the patch 👍!